src/server/emailConfirmation.js

```diff
@@ -9,7 +14,27 @@ export default (request, reply) => {
   if (action === 'supporter' || action === 'petition') {
     confirmRepository.validate(key)
       .then((response) => render(request, reply, 'index'))
-      .catch(() => reply('Invalid key').code(500));
+      .catch((error) => {
+        const statusCode = get(error, 'response.data.error.code', 500);
+        switch (statusCode) {
+          case 404:
+            return render(
+              addServerRequestProps(request, { confirmation: { invalidKey: true } }),
+            reply, 'index');
+          case 400:
+            const description = get(error, 'response.data.error.description', '');
+            if (description.indexOf('Already used') > -1) {
```

Comment:
This might also be a helper, like we did here:
https://github.com/iris-dni/iris-frontend/blob/develop/src/helpers/hasSupportedPetitionAlready.js

`hasConfirmedEmailAlready.js` or something